astroquery/alma/__init__.py

```diff
@@ -19,10 +27,49 @@ class Conf(_config.ConfigNamespace):
     Configuration parameters for `astroquery.alma`.
     """
 
+    # Used if the current discovered host has no Service ID equivalent.  This
+    # is used when overriding the default registry, which assumes to have
+    # just almascience.org as the authority in the Service IDs.
+    default_service_id_auth = 'almascience.org'
+
+    service_uri_map = _arc_service_id_map
+
+    registry_path = _config.ConfigItem(
+        '/reg/resource-caps',
+        'ALMA Registry path')
+
+    tap_service_uri_path = _config.ConfigItem(
+        '/tap',
+        'ALMA TAP ObsCore service URI path')
```

Comment:
This (and two more new configuration items) are currently module constants. Is it now possible for the user to change these values in a meaningful way?

Reply:
It is.  If a service is deployed with a different name than `tap` (e.g. `obscore`), then it would need to be reflected here.  It should really never alter, but this is a single place to change it if need be.

Reply:
Yes.  Each service is identified individually.

Reply:
Is it correct that for `astroquery.alma` there is some fixed set of values this configuration item can take?

Reply:
It's a string as dictated by the implementor, so it could be called anything.  In reality, it will likely only be a single value for the life of the service, unless ALMA makes significant modifications.

Reply:
These have been module level constants so far. Is there a reason why they couldn't remain like that?

Reply:
No reason.  What is the criteria for keeping things inside the module, or in the parent configuration?

Reply:
Having a configuration item for something like the preferred ALMA archive mirror or the username is a good idea because different users are going to want to use different values for those items, but they should not be forced to write boilerplate code in their scripts and they should not be forced to edit the source code because such edits would be easily overwritten whenever they upgrade `astroquery`. The configuration system allows the users to make permanent changes by editing the configuration file instead, and doing it once can be sufficient.

Items such as object identifiers to query should not be in the configuration system because they change very frequently, those should be function arguments instead.

Items such as the new config items implemented in the current version of the pull request should not be configuration items because (if I'm not mistaken) these are determined by the service provider and a user cannot unilaterally change them. Such values should be module constants, like items in question have been so far.